src/number_theory/arithmetic_function.lean

```diff
@@ -231,5 +242,269 @@ instance [comm_ring α] : comm_ring (arithmetic_function α) :=
 { .. arithmetic_function.add_comm_group,
   .. arithmetic_function.comm_semiring }
 
+section pointwise_mul
+
+/-- This is the pointwise product of `arithmetic_function`s. -/
+def pointwise_mul [mul_zero_class α] (f g : arithmetic_function α) :
+  arithmetic_function α :=
+⟨λ x, f x * g x, by simp⟩
+
+@[simp]
+lemma pointwise_mul_apply [mul_zero_class α] {f g : arithmetic_function α} {x : ℕ} :
+  (pointwise_mul f g) x = f x * g x := rfl
+
+lemma pointwise_mul_comm [comm_monoid_with_zero α] (f g : arithmetic_function α) :
+  pointwise_mul f g = pointwise_mul g f :=
+by { ext, simp [mul_comm] }
+
+variable [monoid_with_zero α]
+
+/-- This is the pointwise power of `arithmetic_function`s. -/
+def pointwise_pow (f : arithmetic_function α) (k : ℕ) (kpos : 0 < k) :
+  arithmetic_function α :=
+⟨λ x, (f x) ^ k, by { rw [map_zero], exact zero_pow kpos }⟩
```

Comment:
I wonder if we should add a hack here, and allow for `k = 0` by defining `ppow f 0` to be some random arithmetic function (`id`, or `\zeta`?)
That would allow for nicer notation `f.ppow k`. Lemmas can still require `kpos : 0 < k`, of course.